presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -72,50 +81,73 @@ private RetryDriver()
                 DEFAULT_MAX_RETRY_TIME,
                 Optional.empty(),
                 e -> false,
-                Function.identity());
+                Function.identity(),
+                MoreExecutors.listeningDecorator(executor));
```

Comment:
Why don't you provide the executor of the necessary type?

Reply:
Will address in the next push.